src/Behat/Behat/Output/ServiceContainer/Formatter/JUnitFormatterFactory.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Output\ServiceContainer\Formatter;
+
+use Behat\Testwork\Output\ServiceContainer\OutputExtension;
+use Behat\Testwork\Exception\ServiceContainer\ExceptionExtension;
+use Behat\Testwork\Output\ServiceContainer\Formatter\FormatterFactory;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Behat junit formatter factory.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class JUnitFormatterFactory implements FormatterFactory
```

Comment:
If you make it final, you only have to take care of backwards compatibility for the public API (the usage API). You don't have to worry about breaking things in case someone wants to extend the class.

Besides that, it makes no sense to extend this and if you want to extend this class, there are other better ways of doing so.

Adding a final later on will be hard, as that means a huge break for everyone extending this class. Removing it later on isn't a break, as nobody was extending it (sicne it wasn't possible), so it's just an extra feature.